dotnet-roslyn/analyzers/Audacia.CodeAnalysis.Analyzers/src/Audacia.CodeAnalysis.Analyzers/Rules/MagicNumber/MagicNumberAnalyzer.cs

```diff
@@ -36,22 +44,91 @@ private static void AnalyzeVariable(SyntaxNodeAnalysisContext context)
                 {
-                    AnalyzeBinaryExpression(context, variable, binaryExpression);
+                    AnalyzeBinaryExpression(context, binaryExpression,variable);
                 }
             }
         }
 
-        private static void AnalyzeBinaryExpression(SyntaxNodeAnalysisContext context, VariableDeclaratorSyntax variable, BinaryExpressionSyntax binaryExpression)
+        private static void AnalyzeForSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var variableDeclaration = (ForStatementSyntax)context.Node;
+            AnalyzeVariables(variableDeclaration.Declaration.Variables, context, SyntaxKind.ForStatement);
+        }
+
+        private static void AnalyzeWhileSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var variableDeclaration = (WhileStatementSyntax)context.Node;
+            //AnalyzeVariables(variableDeclaration, context, SyntaxKind.WhileStatement);
+            if(variableDeclaration.Condition is BinaryExpressionSyntax binarySyntax)
+            {
+                AnalyzeBinaryExpression(context,binarySyntax);
+            }
+            
+        }
+
+        private static void AnalyzeIfSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var variableDeclaration = (IfStatementSyntax)context.Node;
+            //AnalyzeVariables(variableDeclaration, context, SyntaxKind.WhileStatement);
+            if (variableDeclaration.Condition is BinaryExpressionSyntax binarySyntax)
+            {
+                AnalyzeBinaryExpression(context, binarySyntax);
+            }
+
+        }
+
+        private static void AnalyzeCaseSwitchLabelSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var variableDeclaration = (CaseSwitchLabelSyntax)context.Node;
+            //AnalyzeVariables(variableDeclaration, context, SyntaxKind.WhileStatement);
+            if (variableDeclaration.Value is LiteralExpressionSyntax numericLiteral)
+            {
+                AnalyzeNumericLiteral(numericLiteral,context);
+            }
+        }
+
+        private static void AnalyzeSwitchStatementSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var variableDeclaration = (SwitchStatementSyntax)context.Node;
+            //AnalyzeVariables(variableDeclaration, context, SyntaxKind.WhileStatement);
+            if (variableDeclaration.Expression is LiteralExpressionSyntax numericLiteral)
+            {
+                AnalyzeNumericLiteral(numericLiteral, context);
+            }
+        }
+
+        private static void AnalyzeNumericLiteral(CSharpSyntaxNode syntaxNode, SyntaxNodeAnalysisContext context,VariableDeclaratorSyntax variable = null)
+        {
+            if (IsAcceptableInteger(syntaxNode))
+            {
+                return;
+            }
+
+            context.ReportDiagnostic(Diagnostic.Create(Rule, syntaxNode.GetLocation(), variable == null ? context.Node.Kind().ToString() : variable.Identifier.Text)) ;
```

Comment:
🪡 Could  `variable == null ? context.Node.Kind().ToString() : variable.Identifier.Text)` be pulled into a variable to make it a little easier to read? (Personal preference)

Reply:
agreed, pulled into a variable

Reply:
Nice one thanks!